pkg/controller/nodes/nodes_controller.go

```diff
@@ -288,3 +351,85 @@ func (r *ReconcileNodes) sendMarkedForTermination(oa *dynatracev1alpha1.OneAgent
 		},
 	})
 }
+
+func (r *ReconcileNodes) reconcileUnschedulableNode(node *corev1.Node, cache *Cache) error {
+	oneAgent, err := r.determineOneAgentForNode(node.Name)
+	if err != nil {
+		return err
+	}
+	if oneAgent == nil {
+		return nil
+	}
+
+	// determineOneAgentForNode  only returns a oneagent object if a node instance is present
+	instance, _ := oneAgent.Status.Instances[node.Name]
+	cachedNode, err := cache.Get(node.Name)
+	if err != nil {
+		if err == ErrNotFound {
+			// If node not found in cache add it
+			cachedNode = CacheEntry{
+				Instance:  oneAgent.Name,
+				IPAddress: instance.IPAddress,
+				LastSeen:  time.Now().UTC(),
+			}
+			err = cache.Set(node.Name, cachedNode)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	return r.markForTermination(cache, oneAgent, instance.IPAddress, node.Name)
+}
+
+func (r *ReconcileNodes) markForTermination(c *Cache, oneAgent *dynatracev1alpha1.OneAgent,
+	ipAddress string, nodeName string) error {
+	cachedNode, err := c.Get(nodeName)
+	if err != nil {
+		return err
+	}
+
+	if isMarkableForTermination(&cachedNode) {
+		err = updateLastMarkedForTerminationTimestamp(c, &cachedNode, nodeName)
+		if err != nil {
+			return err
+		}
+
+		return r.sendMarkedForTermination(oneAgent, ipAddress, cachedNode.LastSeen)
+	}
+	return nil
```

Comment:
It's a Go pattern to exit quickly.

```suggestion
	if !isMarkableForTermination(&cachedNode) {
		return nil
	}

	if err = updateLastMarkedForTerminationTimestamp(c, &cachedNode, nodeName); err != nil {
		return err
	}

	return r.sendMarkedForTermination(oneAgent, ipAddress, cachedNode.LastSeen)
```